pallets/identity/src/lib.rs

```diff
@@ -968,23 +970,22 @@ impl<T: Trait> Module<T> {
             Signatory::Identity(_) => charge_fee()?,
         }
 
-        Self::unsafe_join_identity(target_did, permissions, signer)
+        Self::unsafe_join_identity(target_did, permissions, signer);
+        Ok(())
     }
 
     /// Joins an identity as signer
     pub fn unsafe_join_identity(
         target_did: IdentityId,
         permissions: Permissions,
-        signer: Signatory<T::AccountId>,
-    ) -> DispatchResult {
+        signer: &Signatory<T::AccountId>,
+    ) {
         // Link the secondary key.
-        let sk = SecondaryKey::new(signer, permissions);
+        let sk = SecondaryKey::new(signer.clone(), permissions);
```

Comment:
I don't think it really matters; in the old way you have to `.clone()` either way but in a more top level function, and potentially unnecessarily on errors. In either case, this is actually a memcpy because `AccountId` resolves to a number once generics have been resolved... so I think we can freely clone/pass-by-ref these types of values without thinking about it.